In [84]:
import pandas as pd
import numpy as np
import datetime as dt

In [85]:
# Произведем загрузку датасетов и анализ типов содержащихся данных
founders = pd.read_csv('founders.csv')

In [86]:
#создание функции выводящей все характеристики таблицы на экран
def first_check(dataset):

    print()
    print('Первые 3 строки таблицы')
    display(dataset.head(3))

    print()
    print('Последние 3 строки таблицы')
    display(dataset.tail(3))

    print()
    print('Информация о таблице')   
    print(dataset.info())

    print()
    print('Характеристики чисел в таблице')  
    display(dataset.describe())
      
    print()
    print('Переименование колонок')
def rename(dataset):
    dataset = dataset.rename(columns=lambda c: c.lower())
    return display(dataset.head(3))    

In [87]:
first_check(founders)
rename(founders)


Первые 3 строки таблицы


Name    Birthday Sex             email  \
0     Daniel Carter  1965-01-12   M     law@gmail.com   
1    Samuel Ramirez  1995-12-06   N     pay@gmail.com   
2  Frederick Keller  1960-05-10   F  factor@gmail.com   

                                             Address  \
0  0172 Howard Hill Suite 460\nNorth Peterborough...   
1  41969 Seth Centers Apt. 977\nBakerville, OK 04560   
2                   Unit 4661 Box 5647\nDPO AP 93978   

                      Company          Profession  Salary_USD$  \
0  Campbell, Wang and Russell  Furniture designer         4889   
1  Ross, Wallace and Williams            Best boy         4109   
2    Ward, Adams and Gonzalez  Editor, film/video         4461   

  foundation_date  city_id  city_name  state_id state_code     state_name  \
0      2017-12-26   121610     Medway      1433         MA  Massachusetts   
1      2018-07-08   125110   Richwood      1417         NJ     New Jersey   
2      2019-08-05   124330  Plainview      1452         NY       New York   

                 Funs_club  
0   Alternative Health NYC  
1       Energy Healers NYC  
2  Flourishing Life Meetup


Последние 3 строки таблицы


Name    Birthday Sex              email  \
8056  Lindsey Keller  1979-04-14   F     race@gmail.com   
8057   Brittany Long  1946-03-28   M  hundred@gmail.com   
8058   Tyler Wiggins  1994-03-22   N    allow@gmail.com   

                                                Address  \
8056  23139 Angela Bridge Apt. 572\nMonroeland, ND 4...   
8057           7019 Bradley Locks\nNorth Luke, TX 66610   
8058  47410 Gutierrez Point Suite 455\nPort Mary, NC...   

                          Company                           Profession  \
8056  Ferguson, Johnson and Green  Social research officer, government   
8057    Thomas, Jackson and Brown             Plant breeder/geneticist   
8058  Campbell, Hudson and Garcia                        Acupuncturist   

      Salary_USD$ foundation_date  city_id           city_name  state_id  \
8056         4488      2020-03-31   122089              Monsey      1452   
8057         3717      2020-03-13   127750  Towamensing Trails      1422   
8058         3624      2018-11-11   114297    Clinton Township      1426   

     state_code    state_name Funs_club  
8056         NY      New York       NaN  
8057         PA  Pennsylvania       NaN  
8058         MI      Michigan       NaN


Информация о таблице
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8059 entries, 0 to 8058
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Name             8059 non-null   object
 1   Birthday         8059 non-null   object
 2   Sex              8059 non-null   object
 3   email            8059 non-null   object
 4   Address          8059 non-null   object
 5   Company          8059 non-null   object
 6   Profession       8059 non-null   object
 7   Salary_USD$      8059 non-null   int64 
 8   foundation_date  8059 non-null   object
 9   city_id          8059 non-null   int64 
 10  city_name        8059 non-null   object
 11  state_id         8059 non-null   int64 
 12  state_code       8059 non-null   object
 13  state_name       8059 non-null   object
 14  Funs_club        8034 non-null   object
dtypes: int64(3), object(12)
memory usage: 944.5+ KB
None

Характеристики чисел в таблице


Salary_USD$        city_id     state_id
count   8059.00000    8059.000000  8059.000000
mean    3227.58841  121465.611863  1564.830128
std     1020.37681    7197.056570   660.842859
min     1500.00000  110966.000000  1399.000000
25%     2315.00000  115866.000000  1418.000000
50%     3259.00000  120717.000000  1433.000000
75%     4105.50000  125849.500000  1451.000000
max     4999.00000  153289.000000  4851.000000


Переименование колонок


name    birthday sex             email  \
0     Daniel Carter  1965-01-12   M     law@gmail.com   
1    Samuel Ramirez  1995-12-06   N     pay@gmail.com   
2  Frederick Keller  1960-05-10   F  factor@gmail.com   

                                             address  \
0  0172 Howard Hill Suite 460\nNorth Peterborough...   
1  41969 Seth Centers Apt. 977\nBakerville, OK 04560   
2                   Unit 4661 Box 5647\nDPO AP 93978   

                      company          profession  salary_usd$  \
0  Campbell, Wang and Russell  Furniture designer         4889   
1  Ross, Wallace and Williams            Best boy         4109   
2    Ward, Adams and Gonzalez  Editor, film/video         4461   

  foundation_date  city_id  city_name  state_id state_code     state_name  \
0      2017-12-26   121610     Medway      1433         MA  Massachusetts   
1      2018-07-08   125110   Richwood      1417         NJ     New Jersey   
2      2019-08-05   124330  Plainview      1452         NY       New York   

                 funs_club  
0   Alternative Health NYC  
1       Energy Healers NYC  
2  Flourishing Life Meetup

In [88]:
# переименуем название колонок в нижнем регистре
founders = founders.rename(columns=lambda c: c.lower())

следует сделать следующие преобразования в объекте founders:
- преобразовать строки Birthday и foundation_date в метки даты
- преобразовать числа  city_id  и  state_id  в строки
- преобразовать строки Sex в категории
- столбец Funs_club содержит значения NaN

In [89]:
for column in ['birthday', 'foundation_date']:
    founders[column] = pd.to_datetime(founders[column])

In [90]:
for column in ['city_id', 'state_id']:
    founders[column] = founders[column].astype('str')

In [91]:
founders['sex'] = founders['sex'].astype('category')
founders[:3]

name   birthday sex             email  \
0     Daniel Carter 1965-01-12   M     law@gmail.com   
1    Samuel Ramirez 1995-12-06   N     pay@gmail.com   
2  Frederick Keller 1960-05-10   F  factor@gmail.com   

                                             address  \
0  0172 Howard Hill Suite 460\nNorth Peterborough...   
1  41969 Seth Centers Apt. 977\nBakerville, OK 04560   
2                   Unit 4661 Box 5647\nDPO AP 93978   

                      company          profession  salary_usd$  \
0  Campbell, Wang and Russell  Furniture designer         4889   
1  Ross, Wallace and Williams            Best boy         4109   
2    Ward, Adams and Gonzalez  Editor, film/video         4461   

  foundation_date city_id  city_name state_id state_code     state_name  \
0      2017-12-26  121610     Medway     1433         MA  Massachusetts   
1      2018-07-08  125110   Richwood     1417         NJ     New Jersey   
2      2019-08-05  124330  Plainview     1452         NY       New York   

                 funs_club  
0   Alternative Health NYC  
1       Energy Healers NYC  
2  Flourishing Life Meetup

In [92]:
founders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8059 entries, 0 to 8058
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   name             8059 non-null   object        
 1   birthday         8059 non-null   datetime64[ns]
 2   sex              8059 non-null   category      
 3   email            8059 non-null   object        
 4   address          8059 non-null   object        
 5   company          8059 non-null   object        
 6   profession       8059 non-null   object        
 7   salary_usd$      8059 non-null   int64         
 8   foundation_date  8059 non-null   datetime64[ns]
 9   city_id          8059 non-null   object        
 10  city_name        8059 non-null   object        
 11  state_id         8059 non-null   object        
 12  state_code       8059 non-null   object        
 13  state_name       8059 non-null   object        
 14  funs_club        8034 non-null   object 

In [93]:
# В дальнейшем исследовании будут неинтересны строки с отсутствующими значениями funs_club, поэтому избавимся от них
founders = founders.dropna(subset = ["funs_club"])
founders.head(3)

name   birthday sex             email  \
0     Daniel Carter 1965-01-12   M     law@gmail.com   
1    Samuel Ramirez 1995-12-06   N     pay@gmail.com   
2  Frederick Keller 1960-05-10   F  factor@gmail.com   

                                             address  \
0  0172 Howard Hill Suite 460\nNorth Peterborough...   
1  41969 Seth Centers Apt. 977\nBakerville, OK 04560   
2                   Unit 4661 Box 5647\nDPO AP 93978   

                      company          profession  salary_usd$  \
0  Campbell, Wang and Russell  Furniture designer         4889   
1  Ross, Wallace and Williams            Best boy         4109   
2    Ward, Adams and Gonzalez  Editor, film/video         4461   

  foundation_date city_id  city_name state_id state_code     state_name  \
0      2017-12-26  121610     Medway     1433         MA  Massachusetts   
1      2018-07-08  125110   Richwood     1417         NJ     New Jersey   
2      2019-08-05  124330  Plainview     1452         NY       New York   

                 funs_club  
0   Alternative Health NYC  
1       Energy Healers NYC  
2  Flourishing Life Meetup

In [94]:
founders[['name', 'funs_club']].nunique()

name         7605
funs_club    8034
dtype: int64

В итоге имеем список из 8034 уникальных названий клубов и 7605 уникальных владельцев/основателей (429 человек основали ведут несколько клубов)

Добавим в таблицу расчетную колонку age - возраст основателей на данный момент

In [95]:
from datetime import date
today = date.today()

In [96]:
founders['age'] = today.year - founders['birthday'].dt.year
founders.sample()

name   birthday sex          email  \
1277  Tiffany Davis 2001-02-08   F  ago@gmail.com   

                               address      company       profession  \
1277  Unit 9970 Box 2250\nDPO AE 80007  Hicks Group  Psychotherapist   

      salary_usd$ foundation_date city_id                  city_name state_id  \
1277         4270      2017-06-02  121399  Matanuska-Susitna Borough     1400   

     state_code state_name                                          funs_club  \
1277         AK     Alaska  WisCon in Exile: An SFF book club for WisCon t...   

      age  
1277   22

In [97]:
# основная статитстика по значениям salary и age
founders.describe()

salary_usd$          age
count  8034.000000  8034.000000
mean   3226.384366    52.017675
std    1020.462452    19.623848
min    1500.000000    18.000000
25%    2313.000000    35.000000
50%    3256.000000    52.000000
75%    4104.000000    69.000000
max    4999.000000    86.000000

In [112]:
# наиболее часто встречающийся возраст среди founders - 32, 37 и 51 лет
founders['age'].mode()

0    32
1    37
2    51
Name: age, dtype: int64

In [100]:
# при загрузке сразу переведем некоторые данные в формат datetime
members = pd.read_csv('members.csv', parse_dates = (['Birthday', 'Entry_date']))

In [101]:
first_check(members)
rename(members)


Первые 3 строки таблицы


Name   Birthday Sex                email  \
0  Vanessa Casey 1960-08-08   F  represent@gmail.com   
1   Cheryl Carey 1958-12-11   N      might@gmail.com   
2  Kendra Valdez 1962-06-16   M       nice@gmail.com   

                                             Address  \
0  4295 Kevin Gateway Apt. 257\nPort Leahchester,...   
1    439 Mccann Manors Apt. 741\nJaimebury, KY 56155   
2  72062 Hamilton Trail Suite 863\nPort Nicolas, ...   

                       Company                       Profession  Salary_USD$  \
0             Johnson and Sons  Engineer, manufacturing systems         1884   
1  Duncan, Barnes and Holloway     Trade union research officer         1518   
2     Stokes, Skinner and Moon               Press photographer         1669   

  Entry_date  group_id  city_id   city_name  state_id state_code  \
0 2018-05-24  19927387   122132  Montgomery      1429         WV   
1 2019-11-14  19712846   114446     Colonia      1417         NJ   
2 2018-08-06  10528562   124209    Pinckney      1426         MI   

      state_name  
0  West Virginia  
1     New Jersey  
2       Michigan


Последние 3 строки таблицы


Name   Birthday Sex                email  \
130986  Latasha Alvarez MD 1953-07-07   N  difficult@gmail.com   
130987     Hector Davidson 1945-05-05   N      stuff@gmail.com   
130988     Marissa Krueger 1960-06-15   F       hand@gmail.com   

                                                  Address  \
130986      66877 Thomas Fords\nWest Dianaville, NV 92923   
130987  61513 Jennifer Ports Suite 222\nDavidsoncheste...   
130988  90625 Maxwell Haven Apt. 035\nCandacemouth, WY...   

                            Company       Profession  Salary_USD$ Entry_date  \
130986                  Randall LLC      Interpreter         2579 2018-09-20   
130987  Watson, Murray and Williams    Lexicographer         2890 2018-11-18   
130988               Porter-Jackson  Sales executive         3484 2019-09-08   

        group_id  city_id      city_name  state_id state_code state_name  
130986  23343110   122870  Newton County      1451         MO   Missouri  
130987  10982362   118701        Houston      1400         AK     Alaska  
130988   1454174   117413  Gosper County      1408         NE   Nebraska


Информация о таблице
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130989 entries, 0 to 130988
Data columns (total 15 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   Name         130989 non-null  object        
 1   Birthday     130989 non-null  datetime64[ns]
 2   Sex          130989 non-null  object        
 3   email        130989 non-null  object        
 4   Address      130989 non-null  object        
 5   Company      130989 non-null  object        
 6   Profession   130989 non-null  object        
 7   Salary_USD$  130989 non-null  int64         
 8   Entry_date   130989 non-null  datetime64[ns]
 9   group_id     130989 non-null  int64         
 10  city_id      130989 non-null  int64         
 11  city_name    130989 non-null  object        
 12  state_id     130989 non-null  int64         
 13  state_code   130989 non-null  object        
 14  state_name   130989 non-null  object        
dtypes: datetime6

Salary_USD$      group_id        city_id       state_id
count  130989.000000  1.309890e+05  130989.000000  130989.000000
mean     3246.726435  1.530173e+07  121466.619739    1565.594500
std      1012.220646  9.102480e+06    7179.900295     662.510227
min      1500.000000  6.388000e+03  110966.000000    1399.000000
25%      2367.000000  4.821292e+06  115839.000000    1419.000000
50%      3245.000000  1.892779e+07  120823.000000    1433.000000
75%      4124.000000  2.285461e+07  125883.000000    1451.000000
max      4999.000000  2.637813e+07  153289.000000    4851.000000


Переименование колонок


name   birthday sex                email  \
0  Vanessa Casey 1960-08-08   F  represent@gmail.com   
1   Cheryl Carey 1958-12-11   N      might@gmail.com   
2  Kendra Valdez 1962-06-16   M       nice@gmail.com   

                                             address  \
0  4295 Kevin Gateway Apt. 257\nPort Leahchester,...   
1    439 Mccann Manors Apt. 741\nJaimebury, KY 56155   
2  72062 Hamilton Trail Suite 863\nPort Nicolas, ...   

                       company                       profession  salary_usd$  \
0             Johnson and Sons  Engineer, manufacturing systems         1884   
1  Duncan, Barnes and Holloway     Trade union research officer         1518   
2     Stokes, Skinner and Moon               Press photographer         1669   

  entry_date  group_id  city_id   city_name  state_id state_code  \
0 2018-05-24  19927387   122132  Montgomery      1429         WV   
1 2019-11-14  19712846   114446     Colonia      1417         NJ   
2 2018-08-06  10528562   124209    Pinckney      1426         MI   

      state_name  
0  West Virginia  
1     New Jersey  
2       Michigan

In [102]:
# переименуем название колонок в нижнем регистре
members = members.rename(columns=lambda c: c.lower())
members.sample(3)

name   birthday sex              email  \
100898        Emma Patel 1937-10-03   F     ever@gmail.com   
46978   Edward Christian 1958-11-21   N  control@gmail.com   
130581       Adam Porter 1976-06-16   F    claim@gmail.com   

                                                  address  \
100898   96005 Rich Cape Suite 427\nLake Andrea, SC 93070   
46978   50070 Higgins Lodge Suite 958\nMckenzieside, W...   
130581            3802 Nicole Lane\nWest Hector, TX 05872   

                            company                   profession  salary_usd$  \
100898     Howard, Brown and Chavez       Occupational hygienist         1550   
46978   Landry, Gallagher and Ramos  Corporate investment banker         2986   
130581     Carlson, Silva and Patel      Rural practice surveyor         4336   

       entry_date  group_id  city_id    city_name  state_id state_code  \
100898 2017-07-29     55272   143045   Coto Norte      1449         PR   
46978  2020-01-08  22100910   125652  Saint James      1401         MD   
130581 2020-02-16   3546292   128209   Versailles      1419         KY   

         state_name  
100898  Puerto Rico  
46978      Maryland  
130581     Kentucky

In [103]:
for column in ['city_id', 'state_id', 'group_id']:
    members[column] = members[column].astype('str')

In [113]:
members['sex'] = members['sex'].astype('category')
members[:3]

name   birthday sex                email  \
0  Vanessa Casey 1960-08-08   F  represent@gmail.com   
1   Cheryl Carey 1958-12-11   N      might@gmail.com   
2  Kendra Valdez 1962-06-16   M       nice@gmail.com   

                                             address  \
0  4295 Kevin Gateway Apt. 257\nPort Leahchester,...   
1    439 Mccann Manors Apt. 741\nJaimebury, KY 56155   
2  72062 Hamilton Trail Suite 863\nPort Nicolas, ...   

                       company                       profession  salary_usd$  \
0             Johnson and Sons  Engineer, manufacturing systems         1884   
1  Duncan, Barnes and Holloway     Trade union research officer         1518   
2     Stokes, Skinner and Moon               Press photographer         1669   

  entry_date  group_id city_id   city_name state_id state_code     state_name  \
0 2018-05-24  19927387  122132  Montgomery     1429         WV  West Virginia   
1 2019-11-14  19712846  114446     Colonia     1417         NJ     New Jersey   
2 2018-08-06  10528562  124209    Pinckney     1426         MI       Michigan   

   age  
0   63  
1   65  
2   61

In [114]:
members.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130989 entries, 0 to 130988
Data columns (total 16 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   name         130989 non-null  object        
 1   birthday     130989 non-null  datetime64[ns]
 2   sex          130989 non-null  category      
 3   email        130989 non-null  object        
 4   address      130989 non-null  object        
 5   company      130989 non-null  object        
 6   profession   130989 non-null  object        
 7   salary_usd$  130989 non-null  int64         
 8   entry_date   130989 non-null  datetime64[ns]
 9   group_id     130989 non-null  object        
 10  city_id      130989 non-null  object        
 11  city_name    130989 non-null  object        
 12  state_id     130989 non-null  object        
 13  state_code   130989 non-null  object        
 14  state_name   130989 non-null  object        
 15  age          130989 non-null  int6

In [115]:
members[['name', 'group_id']].nunique()

name        130989
group_id      8035
dtype: int64

Добавим в таблицу расчетную колонку age - возраст основателей на данный момент

In [116]:
members['age'] = today.year - members.birthday.dt.year
members.sample(3)

name   birthday sex                  email  \
16081  Brandi Dickerson 1959-08-06   N  opportunity@gmail.com   
61485     Wendy Padilla 1997-04-28   N         loss@gmail.com   
94687     Ian Hicks PhD 1999-06-05   M     attorney@gmail.com   

                                                 address  \
16081          6356 Kevin Fords\nDanielchester, WY 43492   
61485         834 Kathleen Square\nLake Andrew, AZ 01189   
94687  46330 Danielle Corner Apt. 512\nStephenston, M...   

                          company                      profession  \
16081     Haley, Lopez and Flores               Fish farm manager   
61485   Cooper, Chan and Garrison              Academic librarian   
94687  Rivera, Greene and Sherman  Manufacturing systems engineer   

       salary_usd$ entry_date  group_id city_id     city_name state_id  \
16081         2743 2019-02-23  19196324  121683   Mercersburg     1422   
61485         4094 2019-01-07  18577850  123151    Northfield     1425   
94687         2109 2019-10-19  20209190  120749  Long Prairie     1420   

      state_code    state_name  age  
16081         PA  Pennsylvania   64  
61485         IL      Illinois   26  
94687         MN     Minnesota   24

In [117]:
# основная статитстика по значениям salary и age для members
members.describe()

salary_usd$            age
count  130989.000000  130989.000000
mean     3246.726435      52.259213
std      1012.220646      19.605026
min      1500.000000      18.000000
25%      2367.000000      35.000000
50%      3245.000000      52.000000
75%      4124.000000      69.000000
max      4999.000000      86.000000

In [118]:
# наиболее часто встречающийся возраст среди members - 58 лет
members['age'].mode()

0    58
Name: age, dtype: int64

In [119]:
# загрузим оставшиеся датасеты
hobby_list = pd.read_csv('hobby_list.csv')
categories = pd.read_csv('categories.csv')

In [120]:
first_check(hobby_list)


Первые 3 строки таблицы


group_id                     name  category_id  monthly_fee_USD$
0      6388   Alternative Health NYC           14                19
1     54126       Energy Healers NYC           14                 7
2     67776  Flourishing Life Meetup           14                16


Последние 3 строки таблицы


group_id                         name  category_id  monthly_fee_USD$
8032  25856573  Fuck Yeah Femme Productions           17                19
8033  26158102  Chicago Crossdresser Meetup           17                23
8034  26219043     Corporate Goes Conscious           17                20


Информация о таблице
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8035 entries, 0 to 8034
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   group_id          8035 non-null   int64 
 1   name              8034 non-null   object
 2   category_id       8035 non-null   int64 
 3   monthly_fee_USD$  8035 non-null   int64 
dtypes: int64(3), object(1)
memory usage: 251.2+ KB
None

Характеристики чисел в таблице


group_id  category_id  monthly_fee_USD$
count  8.035000e+03  8035.000000       8035.000000
mean   1.533273e+07    17.397138         11.982078
std    9.106135e+06     9.322519          7.241602
min    6.388000e+03     1.000000          0.000000
25%    4.837647e+06    10.000000          6.000000
50%    1.894228e+07    16.000000         12.000000
75%    2.287862e+07    23.000000         18.000000
max    2.637813e+07    36.000000         24.000000


Переименование колонок


In [121]:
# изменим тип данных на строку
for column in ['group_id', 'category_id']:
    hobby_list[column] = hobby_list[column].astype('str')

In [122]:
hobby_list

group_id                                        name category_id  \
0         6388                      Alternative Health NYC          14   
1        54126                          Energy Healers NYC          14   
2        67776                     Flourishing Life Meetup          14   
3       111855      Hypnosis & NLP NYC - Update Your Brain          14   
4       129277             The Live Food Chicago Community          14   
...        ...                                         ...         ...   
8030  25536270       New York Cuckold Relationships Meetup          17   
8031  25795045  Pagans Paradise NYC - A Haven for Heathens          17   
8032  25856573                 Fuck Yeah Femme Productions          17   
8033  26158102                 Chicago Crossdresser Meetup          17   
8034  26219043                    Corporate Goes Conscious          17   

      monthly_fee_USD$  
0                   19  
1                    7  
2                   16  
3                   13  
4                    2  
...                ...  
8030                16  
8031                21  
8032                19  
8033                23  
8034                20  

[8035 rows x 4 columns]

In [123]:
hobby_list.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8035 entries, 0 to 8034
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   group_id          8035 non-null   object
 1   name              8034 non-null   object
 2   category_id       8035 non-null   object
 3   monthly_fee_USD$  8035 non-null   int64 
dtypes: int64(1), object(3)
memory usage: 251.2+ KB


In [124]:
# заполним колонку monthly_fee_USD$ со значениями NaN нулями
hobby_list['monthly_fee_USD$'] = hobby_list['monthly_fee_USD$'].fillna(0)
hobby_list.head()

group_id                                    name category_id  \
0     6388                  Alternative Health NYC          14   
1    54126                      Energy Healers NYC          14   
2    67776                 Flourishing Life Meetup          14   
3   111855  Hypnosis & NLP NYC - Update Your Brain          14   
4   129277         The Live Food Chicago Community          14   

   monthly_fee_USD$  
0                19  
1                 7  
2                16  
3                13  
4                 2

In [125]:
hobby_list.describe()

monthly_fee_USD$
count       8035.000000
mean          11.982078
std            7.241602
min            0.000000
25%            6.000000
50%           12.000000
75%           18.000000
max           24.000000

In [126]:
categories

category_id               category_name
0             1              Arts & Culture
1             3          Cars & Motorcycles
2             4     Community & Environment
3             5                     Dancing
4             6        Education & Learning
5             9                     Fitness
6            10                Food & Drink
7            11                       Games
8            14          Health & Wellbeing
9            15            Hobbies & Crafts
10           16  Language & Ethnic Identity
11           17                   Lifestyle
12           18                  Book Clubs
13           20               Movies & Film
14           21                       Music
15           22      New Age & Spirituality
16           23        Outdoors & Adventure
17           24                  Paranormal
18           25            Parents & Family
19           27                 Photography
20           28          Religion & Beliefs
21           29            Sci-Fi & Fantasy
22           30                     Singles
23           32         Sports & Recreation
24           33                     Support
25           36                     Writing
26           37               Miscellaneous

In [127]:
categories.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 2 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   category_id    27 non-null     int64 
 1   category_name  27 non-null     object
dtypes: int64(1), object(1)
memory usage: 560.0+ bytes


In [128]:
categories.category_id = categories.category_id.astype('str')

In [129]:
founders.to_csv('founders_1.csv', index = False)

In [130]:
members.to_csv('members_1.csv', index = False)

In [131]:
hobby_list.to_csv('hobby_list_1.csv', index = False)

In [132]:
categories.to_csv('categories_1.csv', index = False)